client/cloudevents/sourceclient.go

```diff
@@ -0,0 +1,315 @@
+package cloudevents
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"time"
+
+	cloudevents "github.com/cloudevents/sdk-go/v2"
+	"github.com/google/uuid"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/klog/v2"
+
+	"open-cluster-management.io/api/client/cloudevents/payload"
+	"open-cluster-management.io/api/client/cloudevents/protocol"
+	"open-cluster-management.io/api/client/cloudevents/types"
+)
+
+type CloudEventSourceClient[T ResourceObject] struct {
```

Comment:
comment please. I think what you mean is this client is used by server but not agent？

Reply:
yes, it is, the source and agent have different new func now